specs/UnhandledKeyPressed.md

```diff
@@ -0,0 +1,248 @@
+        }
+        else if (e.VirtualKey >= 'A' && e.VirtualKey <= 'Z')
+        {
+            Debug.WriteLine($"Ctrl + {(char)e.VirtualKey} not handled by browser is triggered.");
+        }
+    }
+}
+```
+
+# API Details
+
+## Win32 C++
+```idl
+[uuid(053b9a5d-7033-4515-9898-912977d2fde8), object, pointer_default(unique)]
+interface ICoreWebView2Controller : IUnknown {
+  /// Adds an event handler for the `UnhandledKeyPressed` event.
+  /// `UnhandledKeyPressed` will be raised ONLY IF ALL of the following conditions are met:
+  /// 1. The key event has not been marked as handled in the 'AcceleratorKeyPressed' event.
+  /// 2. The key event is not consumed in the DOM (e.g., through calling event.preventDefault()).
+  /// 3. The event is not recognized or taken over by standard browser shortcuts (like ctrl + F, ctrl + P, etc).
+
+  HRESULT add_UnhandledKeyPressed(
+    [in] ICoreWebView2UnhandledKeyPressedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with
+  /// `add_UnhandledKeyPressed`.
+
+  HRESULT remove_UnhandledKeyPressed(
+    [in] EventRegistrationToken token);
+}
+
+[uuid(dc83113b-ce7a-47bb-9661-16b03ff8aac1), object, pointer_default(unique)]
+interface ICoreWebView2UnhandledKeyPressedEventHandler : IUnknown {
+
+  /// Provides the event args for the corresponding event.
+
+  HRESULT Invoke(
+      [in] ICoreWebView2Controller* sender,
+      [in] ICoreWebView2UnhandledKeyPressedEventArgs* args);
+}
+
+[uuid(dd30e20c-d1b3-4e51-9bb6-1be7aaa3ce90), object, pointer_default(unique)]
+interface ICoreWebView2UnhandledKeyPressedEventArgs : IUnknown {
+
+  /// The key event type that caused the event to run.
+
+  [propget] HRESULT KeyEventKind([out, retval] COREWEBVIEW2_KEY_EVENT_KIND* keyEventKind);
+
+  /// The Win32 virtual key code of the key that was pressed or released.  It
+  /// is one of the Win32 virtual key constants such as `VK_RETURN` or an
+  /// (uppercase) ASCII value such as `A`.
+  /// use ICoreWebView2UnhandledKeyPressedEventArgs::get_Modifiers()
+  /// instead of Win32 API ::GetKeyState() to verify whether Ctrl or Alt
+  /// is pressed.
+
+  [propget] HRESULT VirtualKey([out, retval] UINT* virtualKey);
+
+  /// The `LPARAM` value that accompanied the window message.  For more
+  /// information, navigate to [WM_KEYDOWN](/windows/win32/inputdev/wm-keydown)
+  /// and [WM_KEYUP](/windows/win32/inputdev/wm-keyup).
+  /// For visual hosting, only scan code and extended key (16-24bit) are provided.
+  /// For other bits, check out the origin message from ::GetMessage().
+
+  [propget] HRESULT KeyEventLParam([out, retval] INT* lParam);
+
+  /// A structure representing the information passed in the `LPARAM` of the
+  /// window message.
+
+  [propget] HRESULT PhysicalKeyStatus(
+      [out, retval] COREWEBVIEW2_PHYSICAL_KEY_STATUS* physicalKeyStatus);
+
+  /// The `Handled` property will not influence the future WebView
```

Comment:
Does this event need a Handled property? The sample code doesn't show using Handled. At this point, WebView2 has already decided not to do anything with the key. How does setting Handled in this event impact WebView2 behavior?

Reply:
Maybe it can be used when devs register more than one UnhandledKeyPressed handler and they do not want to handle they event on the second handler if the previous handler has marked the event as handled.